go/private/providers.bzl

```diff
@@ -17,6 +17,8 @@ load("@io_bazel_rules_go//go/private:mode.bzl", "mode_string")
 GoLibrary = provider()
 """See go/providers.rst#GoLibrary for full documentation."""
 
+GoPackage = provider()
```

Comment:
Should `GoPackage` be named `GoLibraryData` for consistency with `GoArchiveData`? Or maybe they should be `SourceData` and `CompileData`?

Also, these are never returned as providers, right? Maybe they should be functions that return a `struct`. I suppose that's not really that different but it makes it a little clearer how they're expected to be used.

Reply:
GoPackage came first (and I think it mostly makes sense, it's much more the fundamental values that represent a go package to the rest of Bazel, whereas GoLibrary is the output of the go_library rule, and has a lot more bazel specific stuff in it), and then I could not think of a good name for GoArchiveData.
Generally I think names ending in Data are just lazy, and it's that one that needs renaming, but I could not think of a good name and I really want to get this in to see if we can get k8s compiling so I threw it over the wall....